File: FW 4.0/ESystem/_Extensions/StringExt.vb
Option Explicit On
Option Strict On

Imports System.Runtime.CompilerServices

Namespace Extensions

  Public Module StringExt

    ''' <summary>
    ''' Splits string by another string
    ''' </summary>
    ''' <param name="source"></param>
    ''' <param name="delimiter"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    <Extension()> _
  Public Function Split(ByVal source As String, ByVal delimiter As String) As String()
      Dim ret As String() = source.Split(delimiter)

      Return ret
    End Function

    <Extension()> _
    Public Function RemoveCzechDiakritics(ByRef sourceString As String) As String
      Dim i As Integer
      Dim text As New System.Text.StringBuilder()

      For i = 0 To sourceString.Length - 1
        Select Case sourceString.Chars(i)
          Case "ě"c, "é"c
            text.Append("e")
          Case "Ě"c, "É"c
            text.Append("E")
          Case "ř"c
            text.Append("r")
          Case "Ř"c
            text.Append("R")
          Case "Ť"c
            text.Append("T")
          Case "ť"c
            text.Append("t")
          Case "ý"c
            text.Append("y")
          Case "Ý"c
            text.Append("Y")
          Case "ů"c, "ú"c
            text.Append("u")
          Case "Ú"c
            text.Append("U")
          Case "Í"c
            text.Append("I")
          Case "í"c
            text.Append("i")
          Case "ó"c
            text.Append("o")
          Case "Ó"c
            text.Append("O")
          Case "á"c
            text.Append("a")
          Case "Á"c
            text.Append("A")
          Case "š"c
            text.Append("s")
          Case "Š"c
            text.Append("S")
          Case "ď"c
            text.Append("d")
          Case "Ď"c
            text.Append("D")
          Case "ž"c
            text.Append("z")
          Case "Ž"c
            text.Append("Z")
          Case "č"c
            text.Append("c")
          Case "Č"c
            text.Append("C")
          Case "ň"c
            text.Append("n")
          Case "Ň"c
            text.Append("N")
          Case Else
            text.Append(sourceString.Chars(i))
        End Select
      Next

      Return text.ToString()
    End Function

    <Extension()> _
    Public Function Repeat(ByVal source As String, ByVal count As Integer) As String
      Dim ret As New System.Text.StringBuilder()


      For index As Integer = 1 To count
        ret.Append(source)
      Next

      Return ret.ToString()

    End Function

  End Module

End Namespace

File: FW 3.5/ESystem/_Extensions/StringExt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ESystem.Extensions
{
  public static class StringExt
  {

    public static string[] Split2(this string source, string delimiter)
    {
      string[] ret = source.Split2(delimiter,  StringSplitOptions.None);

      return ret;
    }

    public static string[] Split2(this string source, string delimiter, StringSplitOptions options)
    {
      string[] ret = source.Split(new string[] { delimiter }, options);

      return ret;
    }

    public static string RemoveCzechDiakritics(this string sourceString)
    {
      int i = 0;
      System.Text.StringBuilder text = new System.Text.StringBuilder();

      for (i = 0; i <= sourceString.Length - 1; i++)
      {
        switch (sourceString[i])
        {
          case 'ě':
          case 'é':
            text.Append("e");
            break;
          case 'Ě':
          case 'É':
            text.Append("E");
            break;
          case 'ř':
            text.Append("r");
            break;
          case 'Ř':
            text.Append("R");
            break;
          case 'Ť':
            text.Append("T");
            break;
          case 'ť':
            text.Append("t");
            break;
          case 'ý':
            text.Append("y");
            break;
          case 'Ý':
            text.Append("Y");
            break;
          case 'ů':
          case 'ú':
            text.Append("u");
            break;
          case 'Ú':
            text.Append("U");
            break;
          case 'Í':
            text.Append("I");
            break;
          case 'í':
            text.Append("i");
            break;
          case 'ó':
            text.Append("o");
            break;
          case 'Ó':
            text.Append("O");
            break;
          case 'á':
            text.Append("a");
            break;
          case 'Á':
            text.Append("A");
            break;
          case 'š':
            text.Append("s");
            break;
          case 'Š':
            text.Append("S");
            break;
          case 'ď':
            text.Append("d");
            break;
          case 'Ď':
            text.Append("D");
            break;
          case 'ž':
            text.Append("z");
            break;
          case 'Ž':
            text.Append("Z");
            break;
          case 'č':
            text.Append("c");
            break;
          case 'Č':
            text.Append("C");
            break;
          case 'ň':
            text.Append("n");
            break;
          case 'Ň':
            text.Append("N");
            break;
          default:
            text.Append(sourceString[i]);
            break;
        }
      }

      return text.ToString();
    }

    /// <summary>
    /// Returns string containing "count" times the content of "source" parameter.
    /// </summary>
    /// <param name="source">String to repeat.</param>
    /// <param name="count">Repetition count.</param>
    /// <returns></returns>
    public static string Repeat(this string source, int count)
    {
      System.Text.StringBuilder ret = new System.Text.StringBuilder();


      for (int index = 1; index <= count; index++)
      {
        ret.Append(source);
      }

      return ret.ToString();

    }

  }
}
